core/src/main/scala/org/apache/spark/SparkHadoopWriter.scala

```diff
@@ -143,6 +141,12 @@ class SparkHadoopWriter(jobConf: JobConf)
     taskContext
   }
 
+  protected def newTaskAttemptContext(
```

Comment:
Is this still needed? It seems you replaced the call with a direct call to the constructor in a lot of places.